client/my-sites/site-settings/date-time-format/utils.js

```diff
@@ -22,3 +22,127 @@ export function getLocalizedDate( timezoneString ) {
 		? moment().utcOffset( timezoneString.substring( 3 ) * 60 ) // E.g. "UTC+5" -> "+5" * 60 -> 300
 		: moment.tz( timezoneString );
 }
+
+/**
+ * Mapping object between PHP date() and Moment.js format() tokens.
+ * The commented tokens exist in PHP but have no direct Moment.js equivalent.
+ *
+ * The "S" case is an exception, since it's only used in conjunction with "j":
+ * PHP `date( 'jS' )` returns "1st", which is equivalent to `moment().format( 'Do' )`)
+ *
+ * @see http://php.net/manual/en/function.date.php#refsect1-function.date-parameters
+ * @see http://momentjs.com/docs/#/displaying/format/
+ *
+ * @type {Object}
+ */
+const phpToMomentMapping = {
+	d: 'DD',
+	D: 'ddd',
+	j: 'D',
+	l: 'dddd',
+	N: 'E',
+	// "S" is andled via custom check
+	//S: '',
+	w: 'd',
+	// Moment.js equivalent of "z" (0 based) is "DDD" (1 based), so it must be adjusted
+	//z: 'DDD',
+	W: 'W',
+	F: 'MMMM',
+	m: 'MM',
+	M: 'MMM',
+	n: 'M',
+	// Moment.js has no "t" token equivalent, but a `moment().daysInMonth()` function
+	//t: '',
+	// Moment.js has no "L" token equivalent, but a `moment().isLeapYear()` function
+	//L: '',
+	o: 'Y',
+	Y: 'YYYY',
+	y: 'YY',
+	a: 'a',
+	A: 'A',
+	// Moment.js has no "B" token equivalent, but can be generated nonetheless
+	//B: '',
+	g: 'h',
+	G: 'H',
+	h: 'hh',
+	H: 'HH',
+	i: 'mm',
+	s: 'ss',
+	u: 'SSSSSS',
+	v: 'SSS',
+	e: 'z',
+	// Moment.js has no "I" token, but a `moment().isDST()` function
+	//I: '',
+	O: 'ZZ',
+	P: 'Z',
+	// Moment.js has no "T" token equivalent, but is similar enough to "z"
+	T: 'z',
+	// Moment.js has no "Z" token equivalent, but can be generated nonetheless
+	//Z: '',
+	c: 'YYYY-MM-DDTHH:mm-ssZ',
+	r: 'ddd, DD MMM YYYY HH:mm:ss ZZ',
+	U: 'X',
+};
+
+/**
+ * Convert a PHP datetime format string into a Moment.js one.
+ *
+ * @param {Object} momentDate A Moment.js object of the current date and time.
+ * @param {String} formatString A PHP datetime format string
+ * @return {String} A Moment.js datetime format string
+ */
+export function phpToMomentDatetimeFormat( momentDate, formatString ) {
+	const mappedFormat = formatString.split( '' ).map( ( c, i, a ) => {
```

Comment:
It must work out specifically because those unicode glyph sequences don't match any existing tokens. That's not entirely surprising to me, but I couldn't say I could conclude that from the code that this is the expected behavior 🤷‍♂️